### ISSUE TITLE
An Example

### ISSUE THREAD

**Issue by estock** [pixijs/pixi-spine#29]
Would it be possible to include a basic working example ..  I've tried the pixi V3 examples but can't get them to work any help would be great, I know everything is in heavy flux at the moment

**Comment by Shadowstep33** [comment]
I second this request, and was literally just about to open my own ticket for this. I would love a working example without having to use npm, gulp, or require. I can't figure out how to get it working, unless of course i need to use require.js, but in order to do that I need to completely re-do how all my scripts are being loaded...

A step by step "for dummies" (like myself) guide would be super useful! Btw, I am using pixi 3.0.3

**Comment by ivanpopelyshev** [comment]
1. Download and include latest bin/pixi.js and bin/pixi-spine.js

2. look at http://pixijs.github.io/examples/

**Comment by Shadowstep33** [comment]
Yes, I've already done the following...

> 1. Download and include latest bin/pixi.js and bin/pixi-spine.js . require.js is not required.
 
> 2. look at http://pixijs.github.io/examples/

3.pixi-spine tries to find an atlas file instead of a json (which 404's)

4.Throw error: Uncaught Invalid line: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"

So how do I tell it to load the .json and not .atlas? Do I have to use a .atlas? How should I format said atlas in regards to referencing the layer images by name? Can they differ from how they are labeled in Spine or do the names have to be exact in both the spine .json and the atlas .json? Should I ignore all the v2 docs for Spine since i'm using v3? 

When I only include the .json from spine itself, I don't get any errors; then how does PIXI know what assets to use for this animation? How does it know which assets from the atlas link up to the layers from spine?

**Comment by ivanpopelyshev** [comment]
When you export your spine model into json file, you can also export .atlas file with the same name.
model.json, model.atlas

Are you assembling atlas with different editor (texturepacker) ?

**Comment by Shadowstep33** [comment]
I was using texture packer, but now I am using Spines exported .atlas (as I don't seem to have another option). 

Using the .atlas, I am getting the following error:

```
Skeleton.js:10 Uncaught TypeError: Cannot read property 'length' of undefined
```

**Comment by Shadowstep33** [comment]
I am using 

```
var bunny = new PIXI.spine.Spine(PIXI.spine.Spine.fromAtlas("spider"));
```

and not 

```
dragon = new PIXI.spine.Spine(res.dragon.spineData);
```

because the order I draw things does not match this paradigm. Is there some other way I can access res.dragon.spineData outside of the scope of assetLoader?

**Comment by ivanpopelyshev** [comment]
@Shadowstep33 i know that problem. It means that you pass into new Spine(...) something wrong.

**Comment by ivanpopelyshev** [comment]
Do you load "spider" before it? You can use it only after loader finished its job.

**Comment by ivanpopelyshev** [comment]
```javascript
var bunny = PIXI.spine.Spine.fromAtlas("spider");
```

**Comment by Shadowstep33** [comment]
I load all assets and then draw once they are finished.

I got it working, the following code works for me

```
	loader.use(PIXI.spine.loaders.atlasParser())
	.add('spider','../../spine/spiderdaisy/spider1.json');
	
	loader.on('complete',onAssetsLoaded);

	loader.load();	

function drawStuffs(){
		var spider = new PIXI.spine.Spine.fromAtlas("spider");
		spider.skeleton.setToSetupPose();
		spider.update(0);
		spider.autoUpdate = true;
		
		// create a container for the spine animation and add the animation to it
		var bunny = new PIXI.Container();
		bunny.addChild(spider);

		// once position and scaled, set the animation to play
		spider.state.setAnimationByName(0, 'Idle', true);
}
```

Im omitting a little bit but that seemed to be the important part. I used Spines native Texture Packer functionality, so I did have to use a .atlas.

**Comment by Shadowstep33** [comment]
@ivanpopelyshev  thanks for your help!

**Comment by estock** [comment]
Actually the example referenced worked for me, was able load the data exported from Spine

**Comment by sjlynch** [comment]
Better examples would be nice. I got the dragon animation working after including a trackIndex in the call below. I still don't know what the trackIndex actually is though.

`        animation.state.setAnimationByName(0, "flying", true);
`

**Comment by ivanpopelyshev** [comment]
You can have multiple animations of different parts at the same time, if you assign them to different trackIndices. That syntax completely mirrors vanilla spine runtime, so http://ru.esotericsoftware.com/spine-runtime-terminology applies to it

**Comment by cursedcoder** [comment]
we have examples now, check scripts in `package.json`
